### ISSUE TITLE
[$1000] Attachment preview keeps loading forever when cycle between image and pdf

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#16046]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:

1. Login with any account
2. Go to any chat room
3. Send image as an attachment
4. Send pdf as an attachment
5. Click pdf to show preview
6. Click left and right arrows multiple times to see image and pdf

## Expected Result:
Previews should show correctly

## Actual Result:
 previews keep loading forever and sometimes image shows briefly and disappears (glitch)

## Workaround:
unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [ ] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** 1.2.86-1
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** 

https://user-images.githubusercontent.com/43996225/225682549-71228ec1-4239-4b22-87f0-dde17201af8e.mp4

https://user-images.githubusercontent.com/43996225/225682845-1ab028ec-5f5e-4bbb-97c5-dc04075c6ebc.mp4

**Expensify/Expensify Issue URL:**
**Issue reported by:** @situchan 
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1678644450224539

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01e756be997448cf18</li>
        <li>Upwork Job ID: 1637815865356738560</li>
        <li>Last Price Increase: 2023-05-08</li>
    </ul>
</details>

**Comment by MelvinBot** [comment]
Triggered auto assignment to @stephanieelliott (`Bug`), see https://stackoverflow.com/c/expensify/questions/14418 for more details.

**Comment by stephanieelliott** [comment]
It sounds like this is similar to https://github.com/Expensify/App/issues/15922? However the endless spinner and image flickering seem to be unique to Android so will treat this as separate.

**Comment by MelvinBot** [comment]
Job added to Upwork: https://www.upwork.com/jobs/~01e756be997448cf18

**Comment by MelvinBot** [comment]
Triggered auto assignment to @lschurr (`External`), see https://stackoverflow.com/c/expensify/questions/8582 for more details.

**Comment by MelvinBot** [comment]
Triggered auto assignment to Contributor-plus team member for initial proposal review - @mollfpr (`External`)

**Comment by MelvinBot** [comment]
Triggered auto assignment to @danieldoglas (`External`), see https://stackoverflow.com/c/expensify/questions/7972 for more details.

**Comment by hellohublot** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.

Attachment preview keeps loading forever when cycle between image and pdf

### What is the root cause of that problem?

https://github.com/Expensify/App/blame/e371fc2a956f3f0a3005815e38ab3e0d3a272ebc/patches/react-native-fast-image%2B8.6.3.patch#L174

https://github.com/Expensify/App/pull/13304

https://github.com/DylanVann/react-native-fast-image/issues/975

We can find that our `react-native-fast-image.patch` has modified many things, including obtaining the `cache path` in `onLoadStart` and returning it, but this is `asynchronous`, so `onLoad` will be triggered before `onLoadStart`



### What changes do you think we should make in order to solve the problem?

We can get the image path `synchronously` and return to `onLoadStart` to ensure the correct order

``` java
WritableNativeMap result = new WritableNativeMap();
try {
    String path = requestManager.asFile().load(glideUrl).onlyRetrieveFromCache(true).submit().get().getPath();
    result.putString("cachePath", path);
} catch () {
    result.putNull("cachePath");
}
eventEmitter.receiveEvent(viewId, FastImageViewManager.REACT_ON_LOAD_START_EVENT, result);
```

### What alternative solutions did you explore? (Optional)

But to be honest, we encountered too many problems on `react-native-fast-image`, maybe we should use the `Image` that from `react-native` to replace it, react-native on android will automatically use `Fresco` as a cache, but iOS we You need to add the following code in `AppDelegate` to let the iOS system do the `automatically cache`

``` objective-c
[NSURLCache setSharedURLCache:[[NSURLCache alloc] initWithMemoryCapacity:100 * 1024 * 1024
                                                               diskCapacity: 500 * 1024 * 1024
                                                                   diskPath: nil]];
```

**Comment by mollfpr** [comment]
@tienifr removing `onLoadStart={this.imageLoadingStart}` makes the loading spinner not show and makes the image zoomed after the image loaded.

https://user-images.githubusercontent.com/25520267/226975574-2c33def6-34bf-4f14-9aed-dc0f6e283332.mp4

**Comment by mollfpr** [comment]
@hellohublot Could you show the working result from your proposal?

**Comment by hellohublot** [comment]
## Proposal
[Updated](https://github.com/Expensify/App/issues/16046#issuecomment-1477726116)

@mollfpr Hello. I have modified my solution and added the video link

**Comment by mollfpr** [comment]
@hellohublot Thanks for the update! I still need to get what solution you propose. Could you explain more about that?

**Comment by hellohublot** [comment]
## Proposal
[Updated](https://github.com/Expensify/App/issues/16046#issuecomment-1477726116)

@mollfpr  Hello, I revised my proposal again, if it still confuses you, welcome to ask again, Thank you

**Comment by hellohublot** [comment]
@mollfpr 

https://github.com/DylanVann/react-native-fast-image/blob/9ab80fcd570b7f56da66ab20e52c9a35934067c9/android/src/main/java/com/dylanvann/fastimage/FastImageViewWithUrl.java#L133-L150

No, what actually `sets the picture` is the above piece of code, it will get it from the `cache` first, and if it fails, it will get it from the `network`


https://github.com/Expensify/App/blob/10d3d52e993303db477478d323af925f4d439acf/patches/react-native-fast-image%2B8.6.3.patch#L176-L201

Our `react-native-fast-image.patch.requestManager().onlyRetrieveFromCache().listener(new Listener( onLoadStart() ))` is just simply obtains the `cachePath`, and does not call the `builder.into(ImageView)` method to `set the image`. And I didn't find any code calling `cachePath` in our `all repository`, so this is a piece of `redundant code`, we'd better to revert it

**Comment by mollfpr** [comment]
Hey @thomas-coldwell, sorry to pull you here. I have a question about this [proposal](https://github.com/Expensify/App/issues/16046#issuecomment-1477726116) from @hellohublot, which suggested reverting the below code. Is there will be a problem if we do that?

https://github.com/Expensify/App/blob/b0b31d1b9f424a443a5a475a37ba2029c31a7dbf/patches/react-native-fast-image%2B8.6.3.patch#L171-L201

Also, @cristipaval @Beamanator tagging you guys to confirm if we can update the patch again or if this can be held until we decide what we want to do with `react-native-fast-image`.

Thank you!

**Comment by tienifr** [comment]
> @tienifr removing onLoadStart={this.imageLoadingStart} makes the loading spinner not show and makes the image zoomed after the image loaded.

@mollfpr I'm not seeing the behavior `the loading spinner not show and makes the image zoomed after the image loaded` in the video you attached [in the comment](https://github.com/Expensify/App/issues/16046#issuecomment-1479908648). Am I missing something? Your video looks like it has similar behavior to the current staging app.

**Comment by Beamanator** [comment]
Hey @mollfpr it looks like @cristipaval and @ArekChr just got this PR merged (https://github.com/Expensify/react-native-fast-image/pull/3) to move all of the react-native-fast-image patch-package changes into our Expensify fork 👍 

And I believe they'll be created an App PR soon to remove the patches for that repo, so I think we should wait for that so we can implement the changes in our fork, not in the `patch-package` code 👍 

cc @cristipaval @ArekChr for confirmation 🙏

**Comment by Beamanator** [comment]
Here's the App PR: https://github.com/Expensify/App/pull/16532

**Comment by ArekChr** [comment]
@Beamanator Yes. The patch is going to be removed just testing the latest changes of the fork, soon PR will be opened for review 🚀

**Comment by mollfpr** [comment]
@Beamanator @ArekChr Great to hear that. Thank you for confirming!

@danieldoglas @stephanieelliott Let's hold this to prevent doubling and revisit after https://github.com/Expensify/App/pull/16532 is merged. @Beamanator Is this sounds good?

**Comment by Beamanator** [comment]
Ya that makes the most sense to me too @mollfpr 👍

**Comment by stephanieelliott** [comment]
Oopsie, I auto-doubler got to this before I did -- updated back to $1,000 manually in Upwork

**Comment by MelvinBot** [comment]
Upwork job price has been updated to $1000

**Comment by stephanieelliott** [comment]
still on hold

**Comment by eh2077** [comment]
### Proposal
[Updated](https://github.com/Expensify/App/issues/16046#issuecomment-1482788248)


Hi @stephanieelliott , I think we can consider off-hold this issue as PR https://github.com/Expensify/App/pull/16532 is merged

**Comment by danieldoglas** [comment]
@mollfpr can you test this and check if the behavior is still the same?

**Comment by mollfpr** [comment]
@danieldoglas Yes, it's still reproduced in `v1.3.0-0`



https://user-images.githubusercontent.com/25520267/232117535-2cd9691a-8bd1-4489-9a73-0b526b68413b.mp4

**Comment by danieldoglas** [comment]
Removing hold. @mollfpr let me know your thoughts about the proposals above.

**Comment by mollfpr** [comment]
I haven't got the time, but I'll review the proposal in the morning.

**Comment by mollfpr** [comment]
@eh2077 Can I try your [proposal](https://github.com/Expensify/App/issues/16046#issuecomment-1482788248) locally?

**Comment by eh2077** [comment]
@mollfpr Here's the diff includes the branch for testing.

```diff
diff --git a/package.json b/package.json
index 435e5e6572..c3ca047da8 100644
--- a/package.json
+++ b/package.json
@@ -94,7 +94,7 @@
     "react-native-config": "^1.4.5",
     "react-native-device-info": "^10.3.0",
     "react-native-document-picker": "^8.0.0",
-    "react-native-fast-image": "git+https://github.com/Expensify/react-native-fast-image.git#afedf204bfc253d18f08fdcc5356a2bb82f6a87c",
+    "react-native-fast-image": "git+https://github.com/eh2077/react-native-fast-image.git#a609e9ce2ae87e3e148f461ada29af648565140e",
     "react-native-gesture-handler": "2.9.0",
     "react-native-google-places-autocomplete": "git+https://github.com/Expensify/react-native-google-places-autocomplete.git#e12768f1542e7982d90f6449798f0d6b7f18f192",
     "react-native-haptic-feedback": "^1.13.0",
```

**Comment by danieldoglas** [comment]
@mollfpr how are we with this proposals?

**Comment by MelvinBot** [comment]
📣 It's been a week! Do we have any satisfactory proposals yet? Do we need to adjust the bounty for this issue? 💸

**Comment by mollfpr** [comment]
@danieldoglas Sorry for taking too long. I’ll give my review tonight 🙏

**Comment by mollfpr** [comment]
@danieldoglas I can't reproduce the issue in `v1.3.3-2`.

https://user-images.githubusercontent.com/25520267/234053126-431533a3-b92b-4191-8aa7-3cc69cf9716c.mp4

**Comment by eh2077** [comment]
@mollfpr Here's the video applied the fix FYI

https://user-images.githubusercontent.com/117511920/234066829-4a2f2a31-1b94-4955-85fd-7f46a791355c.mp4

**Comment by danieldoglas** [comment]
@mollfpr which proposal should we proceed with in your opinion?

**Comment by hellohublot** [comment]
@mollfpr  AttachmentCarousel was refactored to be `FlatList`, so steps to reproduce need to change

1. Send 1 picture
2. Send 3 PDF files
3. Click the picture to preview
4. Switch from image to last PDF file, and then switch from last PDF file to image
5. Repeat step 4 several times until you see that the picture is loading infinitely


We can boldly remove `cachePath`
Here is why the author `introduced cachePath`: https://github.com/Expensify/App/pull/13304#issuecomment-1374301735
Here is why the author `removed cachePath`: https://github.com/Expensify/App/pull/13304#issuecomment-1379106062

He explained that he didn’t completely delete `cachePath` because another issue #12893 only wanted to display `loading` when image is loaded from the `network`. When loading from `memory` and `disk`, `loading` don't need be displayed. However, `cachePath` can only get whether there have a `disk cache`, it cannot judge Is there have a `memory cache`, so this idea is wrong

We should remove `cachePath`, and it should be as consistent as possible with `react-native-fast-image` upstream code

**Comment by mollfpr** [comment]
@danieldoglas I need more time to decide this. What @hellohublot explain is interesting.

@hellohublot Could you update your proposal based on your comment?

**Comment by hellohublot** [comment]
@mollfpr 

Thanks, I have updated my [proposal](https://github.com/Expensify/App/issues/16046#issuecomment-1477726116), if you have any concerns or questions, feel free to ask us

**Comment by mollfpr** [comment]
@hellohublot could you attach the fork with your fix so I can test it?

**Comment by hellohublot** [comment]
@mollfpr Sure, You can try this one, `"react-native-fast-image": "git+https://github.com/hellohublot/react-native-fast-image.git#c62408e41f4b91936dcbf187d8d32151b70489d9"`

You can print the `execution order` of `onLoadStart` and `onLoad` in `ImageView.native.js`

**Comment by stephanieelliott** [comment]
Hey @mollfpr have you gotten a chance to test?

**Comment by mollfpr** [comment]
@stephanieelliott Sorry, I miss this issue, and thank you for the ping! I'll test the proposal later.

@hellohublot You are suggesting removing the `cachePath` , so does that mean we don't have any cache image? Is it will introduce this issue again https://github.com/Expensify/App/issues/10792?

**Comment by eh2077** [comment]
Hi @mollfpr, I think it's not good idea to revert back to the [upstream implementation](https://github.com/DylanVann/react-native-fast-image/blob/9ab80fcd570b7f56da66ab20e52c9a35934067c9/android/src/main/java/com/dylanvann/fastimage/FastImageViewWithUrl.java#L127-L129) on Android as it'll always show the loading spinner even the image is cached in the memory. In https://github.com/Expensify/App/pull/13304, people spent a lot of effort to improve it though the solution using `cachePath` is not that perfect.

My [proposal](https://github.com/Expensify/App/issues/16046#issuecomment-1482788248) aims to improve the issue further based what people have tried before. Hope this comment helps.

**Comment by hellohublot** [comment]
@mollfpr 
Sorry, but `cachePath` is the `extra code` we added in `react-native-fast-image`.

it just generates a `flag` to judge whether the picture has been cached on the hard disk, it will not `interfere` with any cache, you can find it in `git blame` in my proposal.

#10792 refers in `react-native-html-render`, replace `react-native-Image`  to `react-native-fast-image`, so it's not related too

We should discard the `redundant code`, should make it consistent with the `original upstream code` and the execution result of iOS.

Hope the following two diffs can help you

[5b9db07#diff-171c9c0c3fbef4818a05438d1976262ffd2ff7f3e8686adc757618735c241694R88](https://github.com/Expensify/App/commit/5b9db070e9e2dec88f756076d5bd8d6f8226f3e8#diff-171c9c0c3fbef4818a05438d1976262ffd2ff7f3e8686adc757618735c241694R88)
[16d7738#diff-171c9c0c3fbef4818a05438d1976262ffd2ff7f3e8686adc757618735c241694L90](https://github.com/Expensify/App/commit/16d773818ef3ac0a791c121a90ec5f2ce44b4fde#diff-171c9c0c3fbef4818a05438d1976262ffd2ff7f3e8686adc757618735c241694L90)

**Comment by MelvinBot** [comment]
📣 It's been a week! Do we have any satisfactory proposals yet? Do we need to adjust the bounty for this issue? 💸

**Comment by mollfpr** [comment]
@danieldoglas the [proposal](https://github.com/Expensify/App/issues/16046#issuecomment-1477726116) from @hellohublot is make sense and work well. I don't see any reason to keep the `cachePath` because we not using it anymore, so we can just revert our implementation https://github.com/Expensify/react-native-fast-image/blob/afedf204bfc253d18f08fdcc5356a2bb82f6a87c/android/src/main/java/com/dylanvann/fastimage/FastImageViewWithUrl.java#L144-L169 to what the upstream implemented https://github.com/DylanVann/react-native-fast-image/blob/9ab80fcd570b7f56da66ab20e52c9a35934067c9/android/src/main/java/com/dylanvann/fastimage/FastImageViewWithUrl.java#L127-L129.

What do you think @danieldoglas?

**Comment by eh2077** [comment]
Hi @mollfpr , can I have your feedback on my comment https://github.com/Expensify/App/issues/16046#issuecomment-1526908538 ?

**Comment by danieldoglas** [comment]
@mollfpr 

I think @hellohublot's proposal looks good, but I think we should test @eh2077 comment's [here](https://github.com/Expensify/App/issues/16046#issuecomment-1526908538). 

Also, let's review @eh2077 's proposal.

**Comment by mollfpr** [comment]
>I think it's not a good idea to revert back to the [upstream implementation](https://github.com/DylanVann/react-native-fast-image/blob/9ab80fcd570b7f56da66ab20e52c9a35934067c9/android/src/main/java/com/dylanvann/fastimage/FastImageViewWithUrl.java#L127-L129) on Android as it'll always show the loading spinner even the image is cached in the memory.

@eh2077 How can I reproduce the loading spinner while the image is cached in memory?

**Comment by eh2077** [comment]
@mollfpr I'd like to help with this but I'm in a hotel with slow network.

**Comment by danieldoglas** [comment]
@mollfpr @eh2077 were you able to test?

**Comment by eh2077** [comment]
@danieldoglas 

I tried but I found that loading spinner issue has been fixed partially in image previewer. The only issue I found  about the loading spinner is that, when clicking an image to preview, both proposals show the loading spinner though I thought my solution should have advantage on this case. See below video

https://user-images.githubusercontent.com/117511920/236893414-a382d1b1-a844-4121-b0eb-1fa227f74a0f.mp4

I'm not sure if it could be a separate issue.

**Comment by mollfpr** [comment]
Thanks @eh2077!

> The only issue I found about the loading spinner is that, when clicking an image to preview, both proposals show the loading spinner though I thought my solution should have advantage on this case.

Can we fix it here? If yes, we can double this issue since it's related to our `react-native-fast-image` fork.

cc @danieldoglas

**Comment by danieldoglas** [comment]
In this case, I'm in favor of adding reporting that as another issue. 


The main issue here was solved and we didn't actually fix it, it was fixed in another PR. Closing this one!

**Comment by situchan** [comment]
Am I eligible for reporting bonus?
The bug was valid at the time of report.

**Comment by stephanieelliott** [comment]
Yep! @situchan created the reporting issue for you here: https://www.upwork.com/jobs/~013fe56fb47ee43e89

**Comment by hellohublot** [comment]
@danieldoglas  Thanks, but I still reproduce from these [steps](https://github.com/Expensify/App/issues/16046#issuecomment-1521178081)

https://github.com/Expensify/App/assets/20135674/b0ba1432-96c6-472d-81b0-3e93e54ff919




Well, let me summarize what are the disadvantages of another proposal

1. A new piece of code has been added. The original `react-native-fast-image` code has been verified by many projects. This code does not have it, and it cannot be more `stable` than the original one.

2. Yes, on the same life cycle, we can ensure that `onLoadStarted` is triggered earlier than `onResourceReady`, but there may be more than one life cycle here, because there is another `FastImageRequestListener`


Another proposal always said I removed `cachePath`, but his own proposal also removed `cachePath`, I think these reasons are ridiculous

**Comment by Christinadobrzyn** [comment]
Hey @mollfpr or @danieldoglas can you confirm this issue is the same https://github.com/Expensify/App/issues/18794

**Comment by mollfpr** [comment]
@Christinadobrzyn I think it's the same root cause.

**Comment by eh2077** [comment]
@danieldoglas Please note that the main issue of this ticket is still reproducible. I'm so sorry for the confusion but my last comment https://github.com/Expensify/App/issues/16046#issuecomment-1538780811 is to reply @mollfpr 's comment https://github.com/Expensify/App/issues/16046#issuecomment-1531521635

> > I think it's not a good idea to revert back to the [upstream implementation](https://github.com/DylanVann/react-native-fast-image/blob/9ab80fcd570b7f56da66ab20e52c9a35934067c9/android/src/main/java/com/dylanvann/fastimage/FastImageViewWithUrl.java#L127-L129) on Android as it'll always show the loading spinner even the image is cached in the memory.
> 
> @eh2077 How can I reproduce the loading spinner while the image is cached in memory?

From which I tried to demonstrate that my proposal should has the advantage to avoid showing the loading spinner when switching images through the Carousel previewer. The current App can avoid showing the loading spinner each time we switch to another image though receiving unnecessary `onLoadStart` events when images are cached in the memory. 

@mollfpr I spent another 1 or 2 hours to test and compare the difference of event triggering between my proposal vs the solution to revert to the original implementation (proposed by @hellohublot). I tested two cases here, switching images through Carousel and tapping to preview image.

According to the log message, for both use cases, my proposal does avoid triggering `onLoadStart` event when the image is cached in the memory. See below video 

https://github.com/Expensify/App/assets/117511920/b29cc76d-a01c-4e67-be12-c64a0de36844

While if reverting to the original implementation, the `onLoadStart` event is always triggered.

https://github.com/Expensify/App/assets/117511920/c82c3943-0300-41df-a597-086737329f52

Although my current proposal also can’t fix the loading spinner of tapping to preview image case, I think it could be a good start to improve the loading spinner issue for memory cached images and, to some degree, it will bring some performance benefits by avoid triggering unnecessary events.

If you also want to test it, this branch https://github.com/eh2077/App/commit/3ed80f3c35f583b76dc84f938630d65388ebfa66 maybe helpful

**Comment by danieldoglas** [comment]
> Please note that the main issue of this ticket is still reproducible.

I see, my bad! I misunderstood it. @stephanieelliott , I'm sorry - I was mistaken about closing this issue before.

I've re-read the whole discussion here. Both proposals are solving the issue described here, which is great!

But I tend to prefer @hellohublot 's proposal since it's reverting our fork back to upstream, keeping us aligned with the stable version - even if it triggers more events. I think we can continue to create changes to the forks as long as they actually solve an issue we're seeing, but if the issue is solved by the stable version, we should go back to it. I am though a little concerned it may cause issues in other usages of the lib through the project, so I suggest we check where else we use it and see if we're not creating any regressions.

@hellohublot, I'm assigning you to the issue.

@eh2077 thank you so much for the investigation into this issue and for the great discussion.

**Comment by danieldoglas** [comment]
I didn't have the right labels here, so fixing it. Sorry for the mess!

**Comment by hellohublot** [comment]
@danieldoglas  Thanks, https://github.com/Expensify/react-native-fast-image/pull/5 upstream pr is ready

**Comment by danieldoglas** [comment]
@mollfpr can you please review the PR? 

Thanks!

**Comment by mollfpr** [comment]
Not overdue

**Comment by danieldoglas** [comment]
@stephanieelliott I'll be OOO for 2 weeks. I'll unassign myself from this issue in case there's progress, can you please add engineering again?

**Comment by stephanieelliott** [comment]
Still waiting on merge of https://github.com/Expensify/App/pull/18997. Will wait to pull un engineer once there is progress

**Comment by stephanieelliott** [comment]
This is still on hold for the other PRs.

**Comment by stephanieelliott** [comment]
Still on hold for of https://github.com/Expensify/App/pull/18997

**Comment by stephanieelliott** [comment]
Still on hold for https://github.com/Expensify/App/pull/18997

**Comment by stephanieelliott** [comment]
Still held on the other issue.

**Comment by stephanieelliott** [comment]
Still blocked on https://github.com/Expensify/App/pull/18997

**Comment by stephanieelliott** [comment]
Ok so it looks like the blocking PR was held on https://github.com/Expensify/react-native-fast-image/pull/6, which was closed in favor of an upstream PR which. has still not been merged. 

However, I just re-tested this and it seems to be resolved now -- perhaps this was fixed by the move to expo-image or https://github.com/Expensify/App/pull/30905 which was recently merged? If so, I think we can pay this out based on the PR that was merged: https://github.com/Expensify/react-native-fast-image/pull/5


--

I was only able to test via simulator, so would love for someone else to test this to make sure it's resolved. Tagging in @kavimuru to see if we can re-test the action steps on this issue to see if it still persists>

![2024-01-03_17-53-46 (1)](https://github.com/Expensify/App/assets/31225194/c1a0b7c5-6cfb-496b-934f-2c8d8306c349)

**Comment by stephanieelliott** [comment]
Tried a few other simulator devices and still unable to repro. Closing this as resolved but feel free to reopen if you can reproduce.

**Comment by situchan** [comment]
@stephanieelliott reporting bonus is pending https://github.com/Expensify/App/issues/16046#issuecomment-1545049279

**Comment by stephanieelliott** [comment]
OH! Yes I still need to issue payment on this one don't I. Ok, considering that this issue resulted in https://github.com/Expensify/react-native-fast-image/pull/5 being created, reviewed and merged, and the PR described in [the proposal ](https://github.com/Expensify/App/issues/16046#issuecomment-1477726116) (https://github.com/Expensify/App/pull/18997) was created and reviewed before being put on hold, I think we should pay out the full amount on this one. 

That would make the payment summary for this issue: 

 - Reporter: @situchan $250 via Upwork (PAID)
 - Contributor: @hellohublot $1,000 via Upwork (offer extended, please accept) 
 - Contributor+: @mollfpr $1,000 NewDot request 

Upwork job is here:

**Comment by stephanieelliott** [comment]
All paid, thank you!

**Comment by JmillsExpensify** [comment]
$1,000 payment approved for @mollfpr based on summary above.
